target_chains/sui/contracts/sources/governance/governance.move

```diff
@@ -7,43 +7,88 @@ module pyth::governance {
     const E_CANNOT_EXECUTE_GOVERNANCE_ACTION_WITH_OBSOLETE_SEQUENCE_NUMBER: u64 = 2;
+    const E_OLD_GUARDIAN_SET_GOVERNANCE: u64 = 3;
+    const E_INVALID_GOVERNANCE_DATA_SOURCE: u64 = 4;
+
+    // this struct does not have the store or key ability so it must be
+    // used in the same txn chain in which it is created
+    struct WormholeVAAVerificationReceipt{
+        payload: vector<u8>,
+        digest: Bytes32,
+        sequence: u64,  // used for replay protection
+    }
+
+    public fun take_payload(receipt: &WormholeVAAVerificationReceipt): vector<u8> {
+        receipt.payload
+    }
+
+    public fun take_digest(receipt: &WormholeVAAVerificationReceipt): Bytes32 {
+        receipt.digest
+    }
+
+    public fun destroy(receipt: WormholeVAAVerificationReceipt) {
+        let WormholeVAAVerificationReceipt{payload: _, digest: _, sequence: _} = receipt;
+    }
+
+   // We define a custom verify_vaa function instead of using wormhole::governance_message::verify_vaa
+   // because that function makes extra assumptions about the VAA payload headers. Pyth uses a
+   // different header format compared to Wormhole, so
+    public fun verify_vaa(
+        pyth_state: &State,
+        verified_vaa: VAA,
+    ): WormholeVAAVerificationReceipt {
+        state::assert_latest_only(pyth_state);
+
+        let vaa_data_source = pyth::data_source::new((vaa::emitter_chain(&verified_vaa) as u64), vaa::emitter_address(&verified_vaa));
+
+        // The emitter chain and address must correspond to the Pyth governance emitter chain and contract.
+        assert!(
+            pyth::state::is_valid_governance_data_source(pyth_state, vaa_data_source),
+            E_INVALID_GOVERNANCE_DATA_SOURCE
+        );
+
+        let digest = vaa::digest(&verified_vaa);
+
+        let sequence = vaa::sequence(&verified_vaa);
+
+        let payload = vaa::take_payload(verified_vaa);
+
+        WormholeVAAVerificationReceipt { payload, digest, sequence }
+    }
 
     /// Execute a governance instruction other than contract upgrade, which is
     /// handled separately in the contract_upgrade.move module.
     public fun execute_governance_instruction(
         pyth_state : &mut State,
-        receipt: DecreeReceipt<GovernanceWitness>,
+        receipt: WormholeVAAVerificationReceipt,
     ) {
         // This capability ensures that the current build version is used.
         let latest_only = state::assert_latest_only(pyth_state);
 
         // Get the sequence number of the governance VAA that was used to
         // generate the receipt.
-        let sequence = governance_message::sequence(&receipt);
+        let sequence = receipt.sequence;
 
         // Require that new sequence number is greater than last executed sequence number.
         assert!(sequence > state::get_last_executed_governance_sequence(pyth_state),
             E_CANNOT_EXECUTE_GOVERNANCE_ACTION_WITH_OBSOLETE_SEQUENCE_NUMBER);
 
+        // consume VAA digest for replay protection
```

Comment:
Is this also necessary? Do we have this logic on other contracts too? I think by just increasing the last executed sequence number, we should be safe

Reply:
May remove the `ConsumedVAAs` struct from `PythState` in a future PR, since it is no longer being used.